ipatests/test_webui/test_automember.py

```diff
@@ -50,16 +49,157 @@
     ],
 }
 
+# Condition types
+INCLUSIVE = 'inclusive'
+EXCLUSIVE = 'exclusive'
+
 
 @pytest.mark.tier1
-class test_automember(UI_driver):
+class TestAutomember(UI_driver):
+
+    AUTOMEMBER_RULE_EXISTS_ERROR = (
+        'Automember rule with name "{}" already exists'
+    )
+
+    def setup(self):
+        super(TestAutomember, self).setup()
+        self.init_app()
+
+    def add_user_group_rules(self, *pkeys, **kwargs):
+        # We implicitly trigger "Add and Add Another" by passing multiple
+        # records to add_record method.
+        # TODO: Create more transparent mechanism to test "Add <entity>" dialog
+        self.add_record(
+            ENTITY,
+            [{
+                'pkey': pkey,
+                'add': [('combobox', 'cn', pkey)],
+            } for pkey in pkeys],
+            facet='searchgroup',
+            **kwargs
+        )
+
+    def add_host_group_rules(self, *pkeys, **kwargs):
+        self.add_record(
+            ENTITY,
+            [{
+                'pkey': pkey,
+                'add': [('combobox', 'cn', pkey)],
+            } for pkey in pkeys],
+            facet='searchhostgroup',
+            **kwargs
+        )
+
+    def add_user(self, pkey, name, surname):
```

Comment:
Does it make sense to move this and similar method from this file to `ui_driver.py`?

Reply:
I'd create separated class for such methods. But currently it is impossible without changing design of base class. UI_driver should be rewritten and I try to not extend it yet.